x-pack/legacy/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -350,6 +370,43 @@ export class VectorLayer extends AbstractLayer {
     };
   }
 
+  async _performInnerJoins(sourceResult, joinStates, updateSourceData) {
+
+    let shouldUpdate = false;
+    for (let i = 0; i < sourceResult.featureCollection.features.length; i++) {
+      const feature = sourceResult.featureCollection.features[i];
+      const oldVisbility = feature.properties[FEATURE_VISIBLE_PROPERTY_NAME];
+      feature.properties[FEATURE_VISIBLE_PROPERTY_NAME] = true;//new visibility will be determined based on join-results
+      for (let j = 0; j < joinStates.length; j++) {
+        const joinState = joinStates[j];
+        const leftInnerJoin = joinState.join;
+        const rightMetricFields = leftInnerJoin.getRightMetricFields();
+
+        let canJoin;
+        if (sourceResult.refreshed || joinState.dataHasChanged) {
+          // Perform join when
+          // - source data changed but join data has not
+          // - join data changed but source data has not
+          // - both source and join data changed
+          canJoin = leftInnerJoin.joinPropertiesToFeature(feature, joinState.propertiesMap, rightMetricFields);
+          shouldUpdate = true;
+        } else {
+          //re-evalute visibility based on join
+          canJoin = leftInnerJoin.canJoin(feature, joinState.propertiesMap);
+        }
+        feature.properties[FEATURE_VISIBLE_PROPERTY_NAME] = feature.properties[FEATURE_VISIBLE_PROPERTY_NAME] && canJoin;
```

Comment:
Change to just `feature.properties[FEATURE_VISIBLE_PROPERTY_NAME] = canJoin `

Reply:
need to only show a feature if _all_ joins have succeeded. to make this more clear, extracting this to separate temp var